tests/target/chains.rs

```diff
@@ -0,0 +1,38 @@
+// Test chain formatting.
+
+fn main() {
+    let a = b.c.d.1.foo(|x| x + 1);
+
+    bbbbbbbbbbbbbbbbbbb.ccccccccccccccccccccccccccccccccccccc.ddddddddddddddddddddddddddd();
+
+    bbbbbbbbbbbbbbbbbbb.ccccccccccccccccccccccccccccccccccccc
+        .ddddddddddddddddddddddddddd
+        .eeeeeeee();
+
+    x().y(|| {
+           match cond() {
+               true => (),
+               false => (),
+           }
+       });
+
+    loong_func()
+        .quux(move || {
```

Comment:
Can we fix this? I don't think we should be inserting unnecessary line breaks like this (It seems to indicate a bug if we are trying to create space where we don't need it - after formatting the closure, we don't need to break the path)